edgelet/edgelet-core/src/module.rs

```diff
@@ -323,16 +325,16 @@ impl SystemInfo {
     }
 }
 
-impl Default for SystemInfo {
-    fn default() -> Self {
-        let kernel = nix::sys::utsname::uname();
+impl SystemInfo {
+    pub fn new() -> nix::Result<Self> {
+        let kernel = nix::sys::utsname::uname()?;
```

Comment:
As unlikely as it is that this fails, I think it would be better to swallow the error and fill in "UNKNOWN" or `null` for all the derived fields, just like you're doing for the `sysname` failure, instead of bailing out completely?

Reply:
d33a376c61bca7f28a7038ca22371fb142062f7d